packages/kbn-search-connectors/lib/fetch_sync_jobs.ts

```diff
@@ -8,82 +8,52 @@
 
 import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
 
-import { CONNECTORS_JOBS_INDEX } from '..';
-import { ConnectorSyncJob, SyncJobType } from '../types/connectors';
+import { ConnectorsAPISyncJobResponse } from '..';
+import { ConnectorSyncJob } from '../types/connectors';
 import { Paginate } from '../types/pagination';
-import { isNotNullish } from '../utils/is_not_nullish';
 
-import { fetchWithPagination } from '../utils/fetch_with_pagination';
-import { isIndexNotFoundException } from '../utils/identify_exceptions';
-
-const defaultResult: Paginate<ConnectorSyncJob> = {
-  _meta: {
-    page: {
-      from: 0,
-      has_more_hits_than_total: false,
-      size: 10,
-      total: 0,
-    },
-  },
-  data: [],
-};
+interface Query {
+  from: number;
+  size: number;
+  connector_id?: string;
+  job_type?: string;
+}
 
 export const fetchSyncJobsByConnectorId = async (
   client: ElasticsearchClient,
-  connectorId: string,
-  from: number,
-  size: number,
+  connectorId?: string,
+  from: number = 0,
+  size: number = 100,
   syncJobType: 'content' | 'access_control' | 'all' = 'all'
 ): Promise<Paginate<ConnectorSyncJob>> => {
-  try {
-    const query =
-      syncJobType === 'all'
-        ? {
-            term: {
-              'connector.id': connectorId,
-            },
-          }
-        : {
-            bool: {
-              filter: [
-                {
-                  term: {
-                    'connector.id': connectorId,
-                  },
-                },
-                {
-                  terms: {
-                    job_type:
-                      syncJobType === 'content'
-                        ? [SyncJobType.FULL, SyncJobType.INCREMENTAL]
-                        : [SyncJobType.ACCESS_CONTROL],
-                  },
-                },
-              ],
-            },
-          };
-    const result = await fetchWithPagination(
-      async () =>
-        await client.search<ConnectorSyncJob>({
-          from,
-          index: CONNECTORS_JOBS_INDEX,
-          query,
-          size,
-          sort: { created_at: { order: 'desc' } },
-        }),
-      from,
-      size
-    );
-    return {
-      ...result,
-      data: result.data
-        .map((hit) => (hit._source ? { ...hit._source, id: hit._id } : null))
-        .filter(isNotNullish),
-    };
-  } catch (error) {
-    if (isIndexNotFoundException(error)) {
-      return defaultResult;
-    }
-    throw error;
+  const body: Query = {
+    from,
+    size,
+  };
+  if (typeof connectorId !== 'undefined') {
```

Comment:
We'd usually just check `!connectorId` or `connectorId !== undefined`. Using typeof is confusing here. 

But I don't really understand why we're checking whether it's defined or not. We can just always assign it, that won't break anything.

Reply:
> We'd usually just check `!connectorId` or `connectorId !== undefined`. Using typeof is confusing here.

Noted.

> But I don't really understand why we're checking whether it's defined or not. We can just always assign it, that won't break anything.

I want to make this function to be able to list all sync jobs, that's why I make `connectorId` optional. I know I keep the function name remain `fetchSyncJobsByConnectorId`, and I know it's confusing. I'm trying to avoid making changes to the usages of this function, and this is a temporary change, we will eventually switch to use the API directly. But I think this does not justify. I will rename the function to `fetchSyncJobs`, and update all the usages accordinglyl.